.github/workflows/publish.yml

```
@@ -9,27 +9,41 @@ on:
       - '**/PKGBUILD'
 
 jobs:
-  publish:
+  setup:
     runs-on: ubuntu-latest
+    outputs:
+      pkgbuilds: ${{ steps.matrix.outputs.pkgbuilds }}
+
     steps:
       - name: Checkout
         uses: actions/checkout@v4.1.6
         with:
           fetch-depth: 0
 
-      - name: Find updated package
+      - name: Find updated packages and construct matrix
+        id: matrix
         run: |
           #!/usr/bin/env bash
           set -euxo pipefail
 
-          echo "pkgbuild=$(git diff --name-only HEAD HEAD~1 "*/.SRCINFO" | head -1 | xargs dirname)" >> $GITHUB_ENV
+          dirs=$(git diff --name-only HEAD HEAD~1 "*/.SRCINFO" | xargs dirname)
+          # Convert dirs that have been changes to a JSON array
+          echo "pkgbuilds=$(printf '["%s"]' "$(echo "$dirs" | sed ':a;N;$!ba;s/\n/","/g')")" >> $GITHUB_OUTPUT
+
+  publish:
+    runs-on: ubuntu-latest
+    needs: [ setup ]
+    strategy:
+      matrix:
+        pkgbuild: ${{fromJSON(needs.setup.outputs.pkgbuilds)}}
 
-      - name: Publish package
+    steps:
+      - name: Publish packages
         uses: KSXGitHub/github-actions-deploy-aur@v2.7.1 # v2.7.1
-        if: ${{ env.pkgbuild != '' }}
+        if: ${{ matrix.pkgbuild != '' }}
         with:
-          pkgname: packages/${{ env.pkgbuild }}
-          pkgbuild: packages/${{ env.pkgbuild }}/PKGBUILD
+          pkgname: packages/${{ matrix.pkgbuild }}
+          pkgbuild: packages/${{ matrix.pkgbuild }}/PKGBUILD
```

We need to remove `packages` here right?
```suggestion
          pkgbuild: ${{ matrix.pkgbuild }}/PKGBUILD
```

---

Yeah i think your right